cloudinit/config/cc_install_hotplug.py

```diff
@@ -59,10 +59,12 @@
 LOG = logging.getLogger(__name__)
 
 
-HOTPLUG_UDEV_PATH = "/etc/udev/rules.d/10-cloud-init-hook-hotplug.rules"
+# 90 to be sorted after 80-net-setup-link.rules which sets ID_NET_DRIVER and
```

Comment:
Just for the record, as this was a bit tricky to find.

[80-net-setup-link.rules](https://github.com/systemd/systemd/blob/main/rules.d/80-net-setup-link.rules) is a systemd-udev upstream rule which is installed by default that calls the [net_setup_setup builtin](https://github.com/systemd/systemd/blob/main/src/udev/udev-builtin-net_setup_link.c) and sets `ID_NET_DRIVER`.

One way to test it is:

```sh
$ SYSTEMD_LOG_LEVEL=debug udevadm test-builtin net_setup_link /sys/class/net/ens5
SELinux enabled state cached to: disabled
Trying to open "/etc/systemd/hwdb/hwdb.bin"...
Trying to open "/etc/udev/hwdb.bin"...
Trying to open "/usr/lib/systemd/hwdb/hwdb.bin"...
Trying to open "/lib/systemd/hwdb/hwdb.bin"...
Trying to open "/lib/udev/hwdb.bin"...
=== trie on-disk ===
tool version:      	249
file size:    	11126818 bytes
header size         	80 bytes
strings        	2374874 bytes
nodes          	8751864 bytes
Load module index
Found cgroup2 on /sys/fs/cgroup/, full unified hierarchy
Found container virtualization none.
Loaded timestamp for '/etc/systemd/network'.
Loaded timestamp for '/run/systemd/network'.
Parsed configuration file /usr/lib/systemd/network/99-default.link
Parsed configuration file /usr/lib/systemd/network/73-usb-net-by-mac.link
Parsed configuration file /run/systemd/network/10-netplan-ens5.link
Created link configuration context.
ID_NET_DRIVER=ena
... more stuff ...
```

Refs:

https://packages.fedoraproject.org/pkgs/systemd/systemd-udev/fedora-rawhide.html
https://packages.debian.org/search?mode=filename&suite=sid&section=all&arch=any&searchon=contents&keywords=80-net-setup-link.rules